### DISCUSSION TITLE
How to fully wait browser load after a form submission by EvaluateScriptAsync

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [cefsharp/CefSharp#4331] [Q&A]
Hello. I am using the below method to submit a form

      var _Command = "document.querySelector(\"body > x > y > z\").shadowRoot.querySelector(\"d > form > ep-button\").shadowRoot.querySelector(\"button\").click()"
      JavascriptResponse scriptResult = await browser.EvaluateScriptAsync(_Command);

However this doesn't wait browser to fully load after command execution. How can I do that?

I am using .NET Core 7 and CefSharp.OffScreen

**Comment by Rand0mRange** [comment]
Use the LoadingStateChanged or FrameLoadEnd events, depending on what you need.

**Reply by FurkanGozukara** [reply]
> Use the LoadingStateChanged or FrameLoadEnd events, depending on what you need.

I cant make sense of it. After executing EvaluateScriptAsync how to use that for waiting full load in the program overflow? 

I tried something like this what you think

```
        private static async Task awaitBrowserLoad(ChromiumWebBrowser browser)
        {
            while(true)
            {
                if (browser.IsLoading == false)
                {
                    browserLoading = false;
                    break;
                }
             await   Task.Delay(100);
            }
        
        }
```

And why there isn't a simple method call to wait until loaded

**Comment by amaitland** [comment]
Already asked at https://stackoverflow.com/questions/74715028/how-to-fully-wait-browser-load-after-a-form-submission-by-evaluatescriptasync-in

Please just ask your question **once** and wait for a response. There's no need to ask multiple times.

**Reply by FurkanGozukara** [reply]
Hello. ty so much for answer but I am getting error when I follow the suggestion you made

![image](https://user-images.githubusercontent.com/19240467/206584310-0d263fd5-86e4-436f-949a-06982c485210.png)

![image](https://user-images.githubusercontent.com/19240467/206584343-65c66460-7965-471c-84be-86a7412ca271.png)

**Reply by amaitland** [reply]
Please review https://idownvotedbecau.se/imageofcode and https://idownvotedbecau.se/imageofanexception/

Have you confirmed the javascript you've executed actually performs a navigation? Try the sample code I provided for testing purposes.
